aidial_adapter_openai/utils/request_classifier.py

```diff
@@ -0,0 +1,26 @@
+from typing import Any
+
+
+def is_request_used_functions_or_tools(data: Any):
+    if (
+        "functions" in data
+        or "function_call" in data
+        or "tools" in data
+        or "tool_choice" in data
+    ):
+        return True
+
+    if "messages" not in data:
+        return False
+
+    for message in data["messages"]:
+        role = message.get("role", None)
```

Comment:
Simply `message.get("role")`. It returns None if the key is missing.

Reply:
fixed